src/dev/file.ts

```diff
@@ -12,11 +12,13 @@ export class File {
   private path: string;
   private relativePath: string;
   private ext: string;
+  private getContentCB: () => string;
 
-  constructor(path: string) {
+  constructor(path: string, getContentCB: () => string = undefined) {
     this.path = resolve(path);
     this.relativePath = relative(process.cwd(), this.path);
     this.ext = extname(this.path);
+    this.getContentCB = getContentCB;
```

Comment:
Do you mind if we drop the `CB` here? Maybe `getVirtualFileContent`?

Reply:
Opted for a different name `fileReader`.